### PR TITLE
Strip role/user mentions (FS-191)

### PR DESCRIPTION
**Pull request by abhiram555**
The issue FS-191 has been fixed in this pr.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -296,9 +296,9 @@
                 <version>3.8.1</version>
                 <configuration>
                     <outputFileName>TotalFreedomMod.jar</outputFileName>
-                    <compilerVersion>11</compilerVersion>
-                    <source>11</source>
-                    <target>11</target>
+                    <compilerVersion>1.8</compilerVersion>
```
Comment (Focusvity) [inline pom.xml:299]: Why? We are now on Java 11, please change it back.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 2]
```diff
@@ -60,6 +62,7 @@
     public ScheduledThreadPoolExecutor RATELIMIT_EXECUTOR;
     public List<CompletableFuture<Message>> sentMessages = new ArrayList<>();
     public Boolean enabled = false;
+    private final Pattern discord_mention_pattern = Pattern.compile("(<@!?([0-9]{16,20})>)");
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:65]: Please follow the `camelCase` naming convention.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 3]
```diff
@@ -408,6 +411,15 @@ public void messageChatChannel(String message)
             message = StringUtils.remove(message, "§");
         }
 
+        // Patch FS-191 start
+        Matcher mention_matcher = this.discord_mention_pattern.matcher(message);
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:415]: Please follow the `camelCase` naming convention.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 4]
```diff
@@ -408,6 +411,15 @@ public void messageChatChannel(String message)
...
+
+        while (mention_matcher.find()) {
+            String mention = mention_matcher.group(1);
+            message = message.replace(mention, "[UserMention-redacted]");
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:419]: I think we should have something better than this.

**Review by Focusvity — CHANGES_REQUESTED** [review]
Formatting and changes are required.

**Comment by Focusvity** [comment]
Just a question, why not loop through the message and see if they start with `<@` and ends with `>` then strip them?

File: pom.xml (hunk 1 above)
Comment (abhiram555) [reply pom.xml:299]: i am using java 8 on my pc so i changed it to java 8. I will change back :D

**Comment by abhiram555** [comment]
> Just a question, why not loop through the message and see if they start with `<@` and ends with `>` then strip them rather than redacting?

i can do it but however at that time the only idea i got is this... but i think this should be fine let me know what you want to do.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java (hunk 4 above)
Comment (abhiram555) [reply src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:419]: we could remove '@' from mention

**Comment by Focusvity** [comment]
> > Just a question, why not loop through the message and see if they start with `<@` and ends with `>` then strip them rather than redacting?
> 
> i can do it but however at that time the only idea i got is this... but i think this should be fine let me know what you want to do.

Please disregard that comment, my brain forgot you can use `String#replace(String, String)`

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 5]
```diff
@@ -60,6 +62,7 @@
     public ScheduledThreadPoolExecutor RATELIMIT_EXECUTOR;
     public List<CompletableFuture<Message>> sentMessages = new ArrayList<>();
     public Boolean enabled = false;
+    private final Pattern DISCORD_MENTION_PATTERN = Pattern.compile("(<@!?([0-9]{16,20})>)");
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:65]: Can we add a check for `<@&ROLE-ID>` please - this pattern is for user mentions.

**Comment by abhiram555** [comment]
> > > Just a question, why not loop through the message and see if they start with `<@` and ends with `>` then strip them rather than redacting?
> > 
> > 
> > i can do it but however at that time the only idea i got is this... but i think this should be fine let me know what you want to do.
> 
> Please disregard that comment, my brain forgot you can use `String#replace(String, String)`

Its okey, is there any other changed that I need to do or are we good to merge?

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 6]
```diff
@@ -408,6 +411,15 @@ public void messageChatChannel(String message)
             message = StringUtils.remove(message, "§");
         }
 
+        // Patch FS-191 start
+        Matcher DISCORD_MENTION_MATCHER = this.DISCORD_MENTION_PATTERN.matcher(message);
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:415]: No need for `this`

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 7]
```diff
@@ -408,6 +411,15 @@ public void messageChatChannel(String message)
...
+
+        while (DISCORD_MENTION_MATCHER.find()) {
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:417]: Formatting - `{` to be on new line

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 8]
```diff
@@ -433,6 +445,15 @@ public void messageAdminChatChannel(String message)
             message = StringUtils.remove(message, "§");
         }
 
+        // Patch FS-191 start
+        Matcher DISCORD_MENTION_MATCHER = this.DISCORD_MENTION_PATTERN.matcher(message);
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:449]: Remove `this`

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 9]
```diff
@@ -433,6 +445,15 @@ public void messageAdminChatChannel(String message)
...
+
+        while (DISCORD_MENTION_MATCHER.find()) {
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:451]: Formatting - `{` on new line

**Review by Focusvity — CHANGES_REQUESTED** [review]
Formatting and changes requested. Can you please remove the comments - it's not needed.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java (hunk 5 above)
Comment (abhiram555) [reply src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:65]: it do work with role id. i just tested it on my local server

**Comment by abhiram555** [comment]
> Formatting and changes requested. Can you please remove the comments - it's not needed.

alrighty

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java (hunk 5 above)
Comment (Focusvity) [reply src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:65]: Can you put these changes on the dev server so I can test them myself? Let me know if you can't and I'll do it.

**Comment by abhiram555** [comment]
Is there any other changes needed?

**Comment by Focusvity** [comment]
> Is there any other changes needed?

I just marked one of the conversations unresolved, can you take a look please.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java (hunk 5 above)
Comment (Focusvity) [reply src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:65]: Just tested and I can confirm it works.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 10]
```diff
@@ -408,6 +411,15 @@ public void messageChatChannel(String message)
             message = StringUtils.remove(message, "§");
         }
 
+
+        Matcher DISCORD_MENTION_MATCHER = this.DISCORD_MENTION_PATTERN.matcher(message);
+
+        while (DISCORD_MENTION_MATCHER.find())
+        {
+            String mention = DISCORD_MENTION_MATCHER.group(1);
+            message = message.replace(mention, mention.replace('@',' '));
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:420]: Can it be just blank? Space doesn't look good.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 11]
```diff
@@ -433,6 +445,15 @@ public void messageAdminChatChannel(String message)
...
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java:454]: Can it be just blank? Space doesn't look good.

**Review by Focusvity — CHANGES_REQUESTED** [review]
Remove spaces and remove `this.` and you're all good.

File: src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java [hunk 12]
```diff
@@ -433,6 +445,15 @@ public void messageAdminChatChannel(String message)
             message = StringUtils.remove(message, "§");
         }
 
+
+        Matcher DISCORD_MENTION_MATCHER = this.DISCORD_MENTION_PATTERN.matcher(message);
+
+        while (DISCORD_MENTION_MATCHER.find())
+        {
+            String mention = DISCORD_MENTION_MATCHER.group(1);
+            message = message.replace(mention, mention.replace('@',''));
```
Comment (abhiram555) [inline src/main/java/me/totalfreedom/totalfreedommod/discord/Discord.java]: Cant do empty char i just looked the error now.

**Review by abhiram555 — COMMENTED** [review]
everything should be good

**Review by Focusvity — APPROVED** [review]
I'm happy with the changes.

**Comment by abhiram555** [comment]
Should i merge?

**Review by Paldiu — APPROVED** [review]

**Pull request merged**